### PR TITLE
Handle a 400 BAD REQUEST correctly

### PR DESCRIPTION
**Pull request by wallner**
Fix a bug that turned a 400 BAD REQUEST response code into a
ConflictException instead of the appropriate BadRequestException.

This pull introduces (MockServer)[http://www.mock-server.com]
as a utility to write more and better tests.

### CODE REVIEW

File: src/test/java/org/osiam/client/OsiamUserServiceTest.java [hunk 1]
```diff
@@ -54,51 +66,51 @@ public void setUp() throws Exception {
...
     @Test(expected = NullPointerException.class)
     public void create_user_with_null_access_token_raises_exception() {
         User newUser = new User.Builder("irrelevant").build();
-
         service.createUser(newUser, null);
+    }
 
-        fail("Exception excpected");
+    @Test(expected = BadRequestException.class)
+    public void invalid_filter_generates_bad_request() {
```
Comment (wallner) [inline src/test/java/org/osiam/client/OsiamUserServiceTest.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/OSIAM/connector4java/file/1934802277/issues/source?bid=3081107&fileBranchId=3103508#l93)

Comment (wallner) [inline src/test/java/org/osiam/client/OsiamUserServiceTest.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/OSIAM/connector4java/file/1937089451/issues/source?bid=3081107&fileBranchId=3103508#l93)

Comment (tkrille) [inline src/test/java/org/osiam/client/OsiamUserServiceTest.java]: Codacy should ignore test code.

Comment (wallner) [inline src/test/java/org/osiam/client/OsiamUserServiceTest.java]: Yes, I already contacted them about it on twitter, but it seems as if you can't just ignore a set of files.

Comment (tkrille) [inline src/test/java/org/osiam/client/OsiamUserServiceTest.java]: You can ignore single files or single issues. I ignored this one, but the commit status is still 'failed'. Anyways, I'll will merge it though.

Comment (tkrille) [inline src/test/java/org/osiam/client/OsiamUserServiceTest.java]: Ooops, It's green now. Hooray!

**Pull request merged**